### PR TITLE
Implement /token_distro_assign_histories webservice, update ethers from v5 to v6

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to https://github.com/Giveth/giveth-dapps-v2/issues/3430

I used Amin's script in this webservice

Screen Shot 1402-09-08 at 16 58 13

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -81,46 +82,51 @@ export const createSmartContractCallAddBatchParams = async (params: {
...
-  })
-  let nonce = await getLastNonceForWalletAddress(givRelayerAddress, network)
-  const rawDatasForHash = []
-  for (let i = 0; i < partNumbers; i++) {
-    const smartContractBatchData = getSmartContractAddBatchesHash(
-      {
-        donationsWithShare: donationsWithShare.slice(i * maxAddressesPerFunctionCall, (i + 1) * maxAddressesPerFunctionCall),
-        nonce
+    }
+    const partNumbers = donationsWithShare.length / maxAddressesPerFunctionCall
```
Comment (jainkrati) [inline src/utils.ts:95]: lets ensure maxAddressesPerFunctionCall is not 0

File: src/givFarm/givFarmService.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          ]
+        }
+     */
+    const args = iface.parseLog(_event as unknown as { topics: string[]; data: string});
+    console.log('args: ', args)
+    // Assuming `amount` is a BigInt.
+    const amountBigInt = BigInt(args.args[2]);
+
+    // Perform division using BigInt and then convert to Number for storing
+    const amountNumber = Number(amountBigInt / 10n ** 18n);
```
Comment (jainkrati) [inline src/givFarm/givFarmService.ts:59]: here and other places, please use ethers.utils.formatEther(weiValue) to convert wei into Ether

**Review by jainkrati — CHANGES_REQUESTED** [review]
once changes are done please DM me in slack to notify.

File: src/utils.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply src/utils.ts:95]: We always call the function like this

https://github.com/Giveth/givback-calculation/blob/staging/src/index.ts#L211-L240


So this line prevents from being `0` or even `NaN`
 Number(maxAddressesPerFunctionCall) || 200

File: src/givFarm/givFarmService.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    Block: ${blockNumber}
+    Transaction Hash: ${transactionHash}
+    Amount: ${amountNumber}
+    -----------------------------------------------
+    `);
+    total += amountBigInt;
+  });
+
+  console.log("###############################################");
+  console.log("Total: ", total / 10n ** 18n);
```
Comment (jainkrati) [inline src/givFarm/givFarmService.ts:76]: here as well?

**Review by jainkrati — APPROVED** [review]

File: src/utils.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply src/utils.ts:99]: You are right, it's better to add validation inside the function

**Pull request merged**